analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantDeclaration.cs

```diff
@@ -270,16 +248,15 @@ private static void ReportRedundantArrayTypeSpecifier(SyntaxNodeAnalysisContext
 
         private static void ReportOnRedundantObjectInitializer(SyntaxNodeAnalysisContext context)
         {
-            var objectCreation = (ObjectCreationExpressionSyntax)context.Node;
+            var objectCreation = ObjectCreationFactory.Create(context.Node);
             if (objectCreation.ArgumentList == null)
             {
                 return;
             }
 
-            if (objectCreation.Initializer != null &&
-                !objectCreation.Initializer.Expressions.Any())
+            if (objectCreation.Initializer != null && !objectCreation.Initializer.Expressions.Any())
```

Comment:
Let's join the conditions of this and the previous `if` into a single `if` statement.